.cascade-code/Chapman.edu/_components/left_nav/left_nav_siblings.vtl

```diff
@@ -0,0 +1,147 @@
+                #else
+                    <a href="${indexPage.link}" class="s-f-idx">
+                        $siblingTitle
+                    </a>
+                    </li>
+                #end
+            #end
+        #end
+    #end
+    </ul>
+#end
+
+## Builds the header for the left nav
+#macro ( buildLeftNavTitle )
+    <div class="leftTitle">
+        <ul>
+            <li class="title">
+                <a href="${parentDir.link}/index" target="_top" title="${_EscapeTool.xml($parentDir.label)}" >
+                    ${_EscapeTool.xml($parentDir.label)}
+                </a>
+            </li>
+        </ul>
+    </div>
+#end
+
+## Sets up the children of a sibling, submenus in the left nav
+#macro ( buildChild $child )
+    #set ( $childTitle = ${_EscapeTool.xml($child.label)} )
+
+    #if ( $child.assetType == 'page' )
+        #set ( $childClass = "c-p" )
+        #set ( $childLink = "${child.link}" )
+    #elseif ( $child.assetType == 'folder' )
+        #set ( $childClass = "c-f" )
+        #set ( $childLink = "${child.link}/index" )
+    #end
+    
+    <li>
+        <a href="${childLink}" title="${childTitle}" class="${childClass}">
+            <span class="bullet">» </span>
+            $childTitle
+        </a>
+    </li>
+#end
+
+## Builds a basic page list item that has no dropdown (no children) in the left nav
+#macro ( buildSiblingPage $sibling )
+    #set ( $siblingTitle = ${_EscapeTool.xml($sibling.label)} )
+
+    #if ( $sibling.path == $currentPage.path )
+        <li class="active">
+    #else
+        <li>
+    #end
+        <a href="${sibling.link}" title="$siblingTitle" class="s-p">$siblingTitle</a>
+    </li>
+#end
+
+## Gets unhidden assets, pages and folders (index pages of folders)
+#macro ( getVisibleChildren $childrenAssets )
+    ## Get all children assets that aren't hidden
+    #foreach ( $asset in $childrenAssets )
+        ## By default, set visibility to false
+        #set ( $visibleChild = false )
+        #if ( $asset.assetType == 'folder')
+            ## Both the folder and index page must be visible
+            #set ( $folderVisible = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+            #set ( $folderIndexPage = $_.locatePage("${asset.path}/index" ) )
+            #set ( $folderIndexVisible = $folderIndexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+            
+            #if ( $folderIndexVisible && $folderVisible )
+                #set ( $visibleChild = true )
+            #end
+        #elseif ( $asset.assetType == 'page' && ($asset.name != 'index') )
+            #set ( $visibleChild = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+        #end
+        
+        #if ( $visibleChild )
+            #set ( $discard = $visibleChildren.add($asset) )
+        #end
+    #end
+#end
+
+##
+## MAIN
+##
+#setParentFolder
+#buildLeftNav
```

Comment:
To echo the sort of MVC practice used by rail, the general pattern I'd recommend here would be:

    setVars()        ## Model in MVC
    buildLeftNav()   ## View in MVC

Secondary macros can be called within these.